tests/lint/git-clang-format.sh

```diff
@@ -58,14 +70,21 @@ fi
 # Print out specific version
 ${CLANG_FORMAT} --version
 
-if [[ ${INPLACE_FORMAT} -eq 1 ]]; then
-    echo "Running inplace git-clang-format against" $1
-    git-${CLANG_FORMAT} --extensions h,mm,c,cc --binary=${CLANG_FORMAT} $1
+if [[ "$INPLACE_FORMAT" == "true" ]]; then
+    echo "Running inplace git-clang-format against $REVISION"
+    git-${CLANG_FORMAT} --extensions h,mm,c,cc --binary=${CLANG_FORMAT} "$REVISION"
     exit 0
 fi
 
-echo "Running git-clang-format against" $1
-git-${CLANG_FORMAT} --diff --extensions h,mm,c,cc --binary=${CLANG_FORMAT} $1 1> /tmp/$$.clang-format.txt
+if [[ "$LINT_ALL_FILES" == "true" ]]; then
+    echo "Running git-clang-format against all C++ files"
+    FILES=$(git ls-files | grep -E '\.(c|cc|cpp|mm|h|hpp)$')
+    git-${CLANG_FORMAT} --diff --extensions h,mm,c,cc --binary=${CLANG_FORMAT} "$REVISION" ${FILES[@]} 1> /tmp/$$.clang-format.txt
```

Comment:
nit: "${FILES[@]}", hopefully this doesn't blow up the max command-line length tho

Reply:
the quotes stop it from word splitting or something or it may be hitting some limit since I see the args are now truncated. In any case this isn't even necessary here since `git-clang-format` will pick up this list anyways since it runs with `REVISION` set to the first commit in the repo

Reply:
didn't we need "${FILES[@]}" to avoid picking up untracked changes in the working tree?

Reply:
answer: no, this wasn't a problem that this PR was trying to fix.